examples/tutorials/api/priors.py

```diff
@@ -0,0 +1,413 @@
+# amplitude value is positive.
+#
+# The weight of the prior again quantifies how strong we want prior to
+# being. Here, we are setting it to 2. This value is only applied if the
+# amplitude value is below zero.
+#
+
+uni_prior_stat_sums = []
+with model_weak_prior.parameters.restore_status():
+    a_scan = np.linspace(-1, 1, 100)
+    for a in a_scan:
+        model_weak_prior.parameters["amplitude"].value = a
+        uni_prior_stat_sums.append(model_weak_prior.parameters.prior_stat_sum())
+
+plt.plot(
+    a_scan,
+    uni_prior_stat_sums,
+    color="tab:orange",
+    linestyle="dashed",
+    label=f"Uniform Prior\n $max={uniform.max.value}$, weight={uniform.weight}",
+)
+plt.xlabel("Amplitude Value [1 / (TeV s cm2)]")
+plt.ylabel("Prior")
+plt.legend()
+plt.show()
+
+
+######################################################################
+# Fitting Multiple Datasets with and without the Prior
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# To showcase how the uniform prior affects the fit results, :math:`100`
+# datasets are created and fitted without and with the prior
+#
+
+results, results_prior = [], []
+N = 100
+dataset2 = dataset.copy()
+for n in range(N):
+    # simulating the dataset
+    dataset2.models = model_weak.copy()
+    dataset2.fake()
+
+    dataset2_prior = dataset2.copy()
+    dataset2_prior.models = model_weak_prior.copy()
+    # fitting without the prior
+    fit = Fit()
+    result = fit.optimize(dataset2)
+    results.append(
+        {
+            "index": result.parameters["index"].value,
+            "amplitude": result.parameters["amplitude"].value,
+        }
+    )
+    # fitting with the prior
+    fit_prior = Fit()
+    result = fit_prior.optimize(dataset2_prior)
+    results_prior.append(
+        {
+            "index": result.parameters["index"].value,
+            "amplitude": result.parameters["amplitude"].value,
+        }
+    )
+
+
+fig, axs = plt.subplots(1, 2, figsize=(7, 4))
+for i, parname in enumerate(["index", "amplitude"]):
+    par = np.array([_[parname] for _ in results])
+    c, bins, _ = axs[i].hist(par, bins=20, alpha=0.5, label="Without Prior")
+    par = np.array([_[parname] for _ in results_prior])
+    axs[i].hist(par, bins=bins, alpha=0.5, color="tab:green", label="With Prior")
+    axs[i].axvline(x=model_weak.parameters[parname].value, color="red")
+    axs[i].set_xlabel(f"Reconstructed spectral\n {parname}")
+    axs[i].legend()
+plt.tight_layout()
+plt.show()
+
+
+######################################################################
+# The distribution of the best-fit amplitudes shows how less best-fit
+# amplitudes have negative values. This also has an effect on the
+# distribution of the best-fit indices. How exactly the distribution
+# changes depends on the weight assigned to the uniform prior. The
+# stronger the weight, the less negative amplitudes.
+#
+# Note that the model parameters uncertainties are, per default, computed
```

Comment:
This is already a heavy tutorial, and I would avoid taking about MINOS etc. You can refer to the fitting tutorial for further details on getting parameter uncertainties.